### PR TITLE
test: add test_volume_wait_for_backing_image_condition

### PR DESCRIPTION
**Pull request by chriscchien**
Add test case `test_volume_wait_for_backing_image_condition`

ref: [7516](https://github.com/longhorn/longhorn/issues/7516)

test [result](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5807/)

### CODE REVIEW

File: manager/integration/tests/test_backing_image.py [hunk 1]
```diff
@@ -648,4 +649,46 @@ def test_volume_wait_for_backing_image_conditino(): # NOQA
...
+        backing_image=backing_img["name"])
+
+    volume2.attach(hostId=lht_host_id)
+
+    for i in range(RETRY_EXEC_COUNTS):
+        volume2 = client.by_id_volume(volume2_name)
+        if volume2.conditions.WaitForBackingImage.status == "True":
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert volume2.conditions.WaitForBackingImage.status == "True"
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_backing_image.py:685]: Is there a way we can check the backingimage is in progress? If yes, do you think below checks make sense?
```
for _ in range(counts):
    If backingimage is in progress:
        assert volume2.conditions.WaitForBackingImage.status == "True"
    break on backingimage completion 

wait_for_volume_healthy
assert volume2.conditions.WaitForBackingImage.status == "False"  
```

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_backing_image.py (hunk 1 above)
Comment (chriscchien) [reply manager/integration/tests/test_backing_image.py]: Updated, utilize `check_backing_image_disk_map_status` to reach the suggestion

test [result](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5847/testReport/tests/test_backing_image/)

File: manager/integration/tests/test_backing_image.py [hunk 2]
```diff
@@ -648,4 +650,46 @@ def test_volume_wait_for_backing_image_conditino(): # NOQA
...
+    volume2 = create_and_check_volume(
+        client, volume_name=volume2_name, size=str(2 * Gi),
+        backing_image=backing_img["name"])
+
+    volume2.attach(hostId=lht_host_id)
+
+    check_backing_image_disk_map_status(client,
+                                        backing_img_name,
+                                        1,
+                                        BACKING_IMAGE_STATE_IN_PROGRESS)
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_backing_image.py:684]: 1. Does all the disk shows `in-progress` status at once or one disk shows `in-progress` and other show `pending` or `waiting`?
2. What's the behavior when backingimage status is complete on only one disk? Does volume utilizes that and become healthy?
3. Is it possible to write the `in-progress` check in if statement to make code more readable?
```
if Backingimage is in progress:
    volume2 = client.by_id_volume(volume2_name)
    assert volume2.conditions.WaitForBackingImage.status == "True"
```

Comment (chriscchien) [reply manager/integration/tests/test_backing_image.py:684]: Hi @khushboo-rancher,
1. At beginning of backingimage start, only 1 diskFileStatusMap exist and the other 2 were empty. from UI can see "File processing is not started".

  <details>
    <summary>In progress backingimage yaml</summary>
  
  ```yaml
 > k get backingimage -A -o yaml
apiVersion: v1
items:
- apiVersion: longhorn.io/v1beta2
  kind: BackingImage
  metadata:
    creationTimestamp: "2024-01-05T03:55:55Z"
    finalizers:
    - longhorn.io
    generation: 4
    labels:
      longhorn.io/component: backing-image
      longhorn.io/managed-by: longhorn-manager
    name: bi-test
    namespace: longhorn-system
    resourceVersion: "45785"
    uid: d7399567-f041-404f-8b84-c671f699e621
  spec:
    checksum: ""
    disks:
      ae46ff66-a2d5-4475-b1e4-2b774b172495: ""
      b387e856-7849-4e1c-8e2c-f9e8cc54a0ee: ""
      b2386520-4a2c-4d8b-8ae1-59a3d424a034: ""
    sourceParameters:
      export-type: qcow2
      volume-name: vol1
    sourceType: export-from-volume
  status:
    checksum: ""
    diskFileStatusMap:
      b2386520-4a2c-4d8b-8ae1-59a3d424a034:
        lastStateTransitionTime: "2024-01-05T03:56:04Z"
        message: ""
        progress: 48
        state: in-progress
    diskLastRefAtMap: {}
    ownerID: ip-172-31-39-108
    size: 2147483648
    uuid: d50fa650
kind: List
metadata:
  resourceVersion: ""
  ```
  </details>

 image

2. After first 1 disk complete(in state `ready`), then the other 2 disks started( have state `in-progress`), in the meantime, volume kept in attaching state, until backingimage on 3 disks all ready, volume become healthy.
image
 
3. Updated!

**Review by khushboo-rancher — APPROVED** [review]
Thanks @chriscchien

**Comment by chriscchien** [comment]
@mergify backport v1.5.x

**Pull request merged**